File: 0014-longest-common-prefix/0014-longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        sort(strs.begin(), strs.end());
        string first = strs[0];
        string last = strs[strs.size() - 1];
        int len = min(first.length(), last.length());
        string acc = "";
        for (int i = 0; i < len; i++){
            if (first[i] == last[i]){
                acc = acc + first[i];
            } else {
                return acc;
            }
        }
        return acc;
    }
};

File: 0014-longest-common-prefix/0014-longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs):
        a = max(strs)
        b = min(strs)
        ans=''
        for i in range(len(b)):
            if a[i]==b[i]:
                ans+=a[i]
            else:
                return ans
        return ans
            



            
        
